hbase-server/src/test/java/org/apache/hadoop/hbase/replication/regionserver/TestRefreshRecoveredReplication.java

```diff
@@ -121,22 +127,25 @@ public void testReplicationRefreshSource() throws Exception {
       table1.put(new Put(r).addColumn(famName, famName, r));
     }
 
-    // kill rs holding table region
-    Optional<RegionServerThread> server = UTIL1.getMiniHBaseCluster().getLiveRegionServerThreads()
-        .stream()
+    // Kill rs holding table region. There are only TWO servers. We depend on it.
+    List<RegionServerThread> rss = UTIL1.getMiniHBaseCluster().getLiveRegionServerThreads();
+    assertEquals(2, rss.size());
+    Optional<RegionServerThread> server = rss.stream()
         .filter(rst -> CollectionUtils.isNotEmpty(rst.getRegionServer().getRegions(tablename)))
         .findAny();
     Assert.assertTrue(server.isPresent());
+    HRegionServer otherServer = rss.get(0).getRegionServer() == server.get().getRegionServer()?
+      rss.get(1).getRegionServer(): rss.get(0).getRegionServer();
     server.get().getRegionServer().abort("stopping for test");
+    // waiting for recovered peer to appear.
+    Replication replication = (Replication)otherServer.getReplicationSourceService();
+    UTIL1.waitFor(60000, () -> !replication.getReplicationManager().getOldSources().isEmpty());
+    // Wait on only one server being up.
     UTIL1.waitFor(60000, () ->
-        UTIL1.getMiniHBaseCluster().getLiveRegionServerThreads().size() == NUM_SLAVES1 - 1);
+      // Have to go back to source here because getLiveRegionServerThreads makes new array each time
+      UTIL1.getMiniHBaseCluster().getLiveRegionServerThreads().size() == NUM_SLAVES1 - 1);
```

Comment:
Instead of waiting on strict equality by count, can you wait for `getLiveRegionServerThreads()` to contain only `otherServer` by identity?

Reply:
I could but skipping as nit if you don't mind.